File: solutions/python/perfect-numbers/1/perfect_numbers.py
def aliquot_sum(n):

    asum = 0
    for i in range(1, n):
        if n % i == 0:
            asum += i
    return asum

def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError('Classification is only possible for positive integers.')
    asum = aliquot_sum(number)
    if asum == number:
        return "perfect"
    if asum < number:
        return 'deficient'
    else:
        return 'abundant'

File: solutions/javascript/perfect-numbers/1/perfect-numbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//
const aliquot = (n) => {
  let sum = 0
  for (let i = 1; i < n; i++) {
    if (n % i === 0) {
      sum += i
    }
  }
  return sum
}
export const classify = (n) => {
  if (n <= 0) {
    throw new Error('Classification is only possible for natural numbers.')
  }
  const asum = aliquot(n)
  if (asum === n) {
    return 'perfect'
  }
  if (asum > n) {
    return 'abundant'
  } else {
    return 'deficient'
  }
};
